src/Carbon/Traits/Test.php

```diff
@@ -128,12 +128,14 @@ public static function setTestNowAndTimezone($testNow = null, $tz = null)
      */
     public static function withTestNow($testNow, $callback)
     {
+        $original = static::now();
```

Comment:
For the record, using `static::now()` here will return n instance even if it was `null` before. So if the time was running (no mock) before, then after `withTestNow()` you won't go back to "real time" running, you will go back to frozen time of the exact moment before executing `withTestNow()`.

Here instead, to get the behavior you want, the right method to call would be `static::getTestNow()`.